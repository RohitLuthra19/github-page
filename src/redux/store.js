import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";

import rootReducer from "./rootReducer";
import { getProfileWatcherSaga } from "./user/sagas-get-user-profile";
import { getRepositoriesWatcherSaga } from "./user/sagas-get-repositories"


const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  composeEnhancers(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(getProfileWatcherSaga);
sagaMiddleware.run(getRepositoriesWatcherSaga);

export default store;