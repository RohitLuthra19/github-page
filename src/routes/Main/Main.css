.repo-list {
  padding: 0;
  margin: 0;
}

.search-form {
  margin: 16px 0;
}

.search-box {
  width: 100%;
  margin-right: 16px;
  padding: 5px 12px;
  font-size: 14px;
  line-height: 20px;
  color: #24292e;
  vertical-align: middle;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #e1e4e8;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(225,228,232,.2);
}

.btn-primary {
  width: 90px;
  color: #fff;
  background-color: #2ea44f;
  border-color: rgba(27,31,35,.15);
  box-shadow: 0 1px 0 rgba(27,31,35,.1), inset 0 1px 0 hsla(0,0%,100%,.03);
}

.repo-icon {
  vertical-align: middle;
  color: #fff;
  fill: #fff;
}

.md-flex {
  display: flex;
}

.md-flex-wrap {
  flex-wrap: nowrap;
}
  
@media (max-width: 600px) {
  .md-flex-wrap {
    flex-wrap: wrap;
  }
}