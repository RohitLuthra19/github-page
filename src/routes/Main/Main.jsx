import React from "react";
import { connect } from "react-redux";

import ListItem from "../../components/ListItem";
import "./Main.css";
import { getRepositories } from "../../redux/user/reducer";
import RepoIcon from "../../assets/repo-icon";

export class Main extends React.PureComponent {
  state = {
    searchInput: "",
  };

  handleChange = (event) => {
    const { value } = event.target;
    this.setState({ searchInput: value });
  };

  render() {
    const { repos, fetching } = this.props;
    const { searchInput } = this.state;

    if(fetching) {
      return <>Loading...</>
    } else return (
      <>
        <form
          aria-label="Repositories"
          role="search"
          className="search-form md-flex md-flex-wrap"
        >
          <input
            type="search"
            id="repos-filter"
            name="search-box"
            placeholder="Find a repository…"
            autoComplete="off"
            aria-label="Find a repository…"
            value={searchInput}
            className="search-box"
            onChange={this.handleChange}
          />
          <div className="md-flex">
            <button className="btn btn-primary">
              <RepoIcon className="repo-icon" /> New
            </button>
          </div>
        </form>
        <ul className="repo-list">{this.renderRepositories(repos)}</ul>
      </>
    );
  }

  componentDidMount() {
    this.props.getRepositories();
  }

  componentDidUpdate(prevProps) {}

  ///////////////////////////////////////////////////////////////////////
  //  RENDER METHODS
  ///////////////////////////////////////////////////////////////////////
  renderRepositories(repos) {
    const { searchInput } = this.state;
    return repos
      .filter((item) =>
        item?.name?.toLowerCase().includes(searchInput?.toLowerCase())
      )
      .map((repo) => {
        return <ListItem key={repo?.id} data={repo} />;
      });
  }
}

///////////////////////////////////////////////////////////////////////
//  REDUX CONNECTION
///////////////////////////////////////////////////////////////////////
function mapStateToProps(state) {
  const { user } = state;
  const userToJS = user.toJS();

  return {
    repos: userToJS.user.repos,
    fetching: userToJS.user.fetching
  };
}

// don't need mapDispatchToProps b/c we are using action creators
export default connect(mapStateToProps, {
  getRepositories,
})(Main);
