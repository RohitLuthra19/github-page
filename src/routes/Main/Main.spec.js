import React from 'react';
import Main from './Main';
import '../../setupTests';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import { shallow } from 'enzyme';

describe('Main renders',() => {
  it('renders', () => {
    const wrapper = shallow(<Main store={store} />);
    expect(wrapper.exists()).toBe(true);
  })

  it('Main component created', () => {
    const rendered = renderer.create(
      <Main store={store} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  })
})