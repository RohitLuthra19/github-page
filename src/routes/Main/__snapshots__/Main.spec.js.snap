// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders Main component created 1`] = `"Loading..."`;
